app/models/user.rb

```diff
@@ -183,7 +183,7 @@ def direct_filters
 
   def expire_caches
     if login_changed?
-      self.works.each{ |work| work.touch }
+      self.works.each{ |work| work.expire_caches }
```

Comment:
Just checking -- will this also update the `updated_at` timestamp on the work, just like `touch` did? If so, I think this PR is :+1:, but wouldn't mind another opinion.